tests/components/notion/test_config_flow.py

```diff
@@ -112,3 +112,8 @@ async def test_reauth(
     assert result["type"] == data_entry_flow.FlowResultType.ABORT
     assert result["reason"] == "reauth_successful"
     assert len(hass.config_entries.async_entries()) == 1
+
+    # Block to ensure the setup_config_entry fixture does not
+    # get undone before hass is shutdown so we do not try
+    # to setup the config entry via reload.
+    await hass.async_block_till_done()
```

Comment:
We normally put the block till done directly after the function call that spawns the task.
```suggestion
    # Block to ensure the setup_config_entry fixture does not
    # get undone before hass is shutdown so we do not try
    # to setup the config entry via reload.
    await hass.async_block_till_done()

    assert result["type"] == data_entry_flow.FlowResultType.ABORT
    assert result["reason"] == "reauth_successful"
    assert len(hass.config_entries.async_entries()) == 1    
```